Assets/Scripts/Client/Systems/UpdateMaterialComponentData.cs

```diff
@@ -11,31 +11,37 @@ namespace PropHunt.Client.Systems
     /// When a material of an entity needs to be updated, this system will update the render mesh.
     /// </summary>
     [UpdateInGroup(typeof(ClientSimulationSystemGroup))]
-    public class UpdateMaterialSystem : ComponentSystem
+    public class UpdateMaterialSystem : SystemBase
     {
         protected override void OnUpdate()
         {
             var ecb = EntityManager.World.GetOrCreateSystem<EndSimulationEntityCommandBufferSystem>().CreateCommandBuffer();
+            var getSharedMaterialsFromEntity = GetComponentDataFromEntity<SharedMaterialData>(true);
+            var sharedMaterialEntity = EntityManager.CreateEntityQuery(typeof(SharedMaterialData)).GetSingletonEntity();
 
-            Entities.WithNone<NetworkStreamInGame>().ForEach((Entity ent, ref UpdateMaterialComponentData updateMatTag, ref MaterialIdComponentData materialId) =>
+            Entities.WithNone<NetworkStreamInGame>()
+            .WithReadOnly(getSharedMaterialsFromEntity)
+            .ForEach((Entity ent, in UpdateMaterialComponentData updateMatTag, in MaterialIdComponentData materialId) =>
             {
                 // We only want to update this entity once, so this tag will be removed afterwards.
-                PostUpdateCommands.RemoveComponent<UpdateMaterialComponentData>(ent);
+                ecb.RemoveComponent<UpdateMaterialComponentData>(ent);
+                var sharedMaterialData = getSharedMaterialsFromEntity[sharedMaterialEntity];
```

Comment:
Instead of adding and using an entity for UpdateMaterialComponentData, could we do something like a [change filter](https://docs.unity3d.com/Packages/com.unity.entities@0.16/manual/ecs_entity_query.html#change-filters)? It allows you to filter entities (by chunks) for when the data changed. 

That way we can change them when they are initially setup and then support changing color in future updates :o 

This would be really good to try out for things like the props updating